config/types_test.go

```diff
@@ -0,0 +1,190 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServiceRegistrationToConsulType(t *testing.T) {
```

Comment:
It would be super helpful to have a test that the types round trip rather than just one way. In the past I've done this with property based test, but it probably isn't practical in Go.

This is a random thought and probably not actionable

Reply:
Yeah, I'm talking about `JSON -> Our Types -> Consul`. In the past, I've found tons of bugs with tests like this. I had no idea 'testing/quick' existed. That's really cool.